tck-spec/src/test/kotlin/io/rsocket/spec/StepDefinitions.kt

```diff
@@ -0,0 +1,93 @@
+package io.rsocket.spec
+
+import io.cucumber.datatable.DataTable
+import io.cucumber.java.en.Given
+import io.cucumber.java.en.Then
+import io.cucumber.java.en.When
+import io.netty.bootstrap.ServerBootstrap
+import io.netty.buffer.ByteBuf
+import io.netty.channel.*
+import io.netty.channel.nio.NioEventLoopGroup
+import io.netty.channel.socket.SocketChannel
+import io.netty.channel.socket.nio.NioServerSocketChannel
+import io.rsocket.frame.SetupFlags
+import io.rsocket.frame.SetupFrame
+import io.rsocket.frame.shared.FrameHeader
+import io.rsocket.frame.shared.KeepAlive
+import io.rsocket.frame.shared.Version
+import java.util.concurrent.TimeUnit.MILLISECONDS
+import kotlin.time.toDuration
+
+class StepDefinitions {
+    private val initializer = ServerChannelInitializer()
+    private lateinit var server: Server
+    private lateinit var expected: SetupFrame
+    private lateinit var actual: SetupFrame
+
+    @Given("server listening {port} port")
+    fun successful_connection_establishment(port: Int) {
+        server = Server(port)
+        server.run()
+    }
+
+    @When("client sends SETUP frame")
+    fun run(data: DataTable) {
+        val map = data.asMap<String, String>(String::class.java, String::class.java);
+        expected = SetupFrame(
+            FrameHeader(map["stream-id"]!!.toInt(), SetupFlags(false, false, false)),
+            Version(map["major-version"]!!.toInt(), map["minor-version"]!!.toInt()),
+            KeepAlive(
+                map["keep-alive"]!!.toInt().toDuration(MILLISECONDS),
+                map["max-life-time"]!!.toInt().toDuration(MILLISECONDS))
+        )
+    }
+
+    @Then("server does not close connection")
+    fun finishTest() {
+        assert(expected == actual)
+    }
+}
+
+class Server(private val port: Int) {
+    @Throws(Exception::class)
```

Comment:
`Throws` annotation isn't needed, if you not use both kotlin and java in one project